depends/packages/boost.mk

```diff
@@ -18,14 +18,14 @@ $(package)_config_opts_linux=target-os=linux threadapi=pthread runtime-link=shar
 $(package)_config_opts_freebsd=cxxflags=-fPIC
 $(package)_config_opts_darwin=target-os=darwin runtime-link=shared
 $(package)_config_opts_mingw32=target-os=windows binary-format=pe threadapi=win32 runtime-link=static
-$(package)_toolset_$(host_os)=clang
-$(package)_toolset_darwin=clang
 $(package)_config_opts_x86_64=architecture=x86 address-model=64
 $(package)_config_opts_i686=architecture=x86 address-model=32
 $(package)_config_opts_aarch64=address-model=64
 $(package)_config_opts_armv7a=address-model=32
 ifneq (,$(findstring clang,$($(package)_cxx)))
-   $(package)_toolset_$(host_os)=clang
+$(package)_toolset_$(host_os)=clang
+else
+$(package)_toolset_$(host_os)=gcc
```

Comment:
I can't tell whether this change is correct for us. We want to allow overriding CXX to g++, but we shouldn't do so by default.

Reply:
For us this does end up correct, because we set `$(package)_cxx` to `clang` explicitly. But if someone were to try overriding to use GCC, this would at least give them the correct behaviour for this package.